### PR TITLE
Implement supporting DHCP

### PR DESCRIPTION
**Pull request by MykolaSolyanko**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -100,7 +100,78 @@ To launch VM image, use the following script:
...
+Follow the script output instruction to attach to nodes consoles. -->
+
+## Create VM Image Archive
+
+To create an archive of the VM images, use the following command:
+
+```sh
+yocto/meta-aos-vm/scripts/aos_vm.sh archive -m -s <number_of_secondary_nodes> -o <output_path>/aos-vm.tar.gz
+
+options:
```
Comment (al1img) [inline README.md:112]: options explanation should be out of sh code block.

File: README.md [hunk 2]
```diff
@@ -100,7 +100,78 @@ To launch VM image, use the following script:
...
+This command will create a tar.gz archive containing one main node and two secondary nodes in the specified output path.
+
+## Generate VM Image
+
+To generate VMDK disks for the VM images, use the following command:
+
+```sh
+yocto/meta-aos-vm/scripts/aos_vm.sh create -m -s <number_of_secondary_nodes> -o <output_path>
+
+options:
```
Comment (al1img) [inline README.md:133]: ditto

File: README.md [hunk 3]
```diff
@@ -100,7 +100,78 @@ To launch VM image, use the following script:
...
+This command will create VMDK files for one main node and two secondary nodes in the specified output path.
+
+## Run VM Image
+
+To run the VM images, use the following command:
+
+```sh
+yocto/meta-aos-vm/scripts/aos_vm.sh run -f <file_or_folder>
+
+options:
```
Comment (al1img) [inline README.md:154]: ditto

File: scripts/run_aos_vm.sh [hunk 4]
```diff
@@ -77,33 +76,45 @@ if [ -z "$(ip link show | grep $bridge_name)" ]; then
...
         continue
     fi
 
-    echo "Starting $node..."
+    # Extract node name from the filename
+    filename=$(basename -- "$node_image")
+    node=$(echo "$filename" | sed 's/aos-vm-\(.*\)-genericx86-64.wic.vmdk/\1/')
 
-    start_node "$node" "$node_image" "${nodes_cpu[$i]}" "${nodes_ram[$i]}" "${nodes_mac[$i]}" &
+    cpu="1"
```
[the remark this answers is not in the document]
Comment (MykolaSolyanko) [reply scripts/run_aos_vm.sh]: Done

File: scripts/create_aos_vm.sh [hunk 5]
```diff
@@ -2,42 +2,90 @@
...
+    local image_path="$3"
+
+    if [ ! -f "$node_image" ]; then
+        echo "Image file $node_image not found."
+        return 1
+    fi
+
+    echo "Found image $node_image"
+    echo "Converting $node_image to vmdk..."
+    qemu-img convert -p -f raw -O vmdk "${node_image}" "$image_path/${vmdk_name}"
```
[the remark this answers is not in the document]
Comment (MykolaSolyanko) [reply scripts/create_aos_vm.sh]: Done

File: scripts/create_aos_vm.sh [hunk 6]
```diff
@@ -2,42 +2,90 @@
...
+
+secondary_count=1
+
+for arg in "$@"; do
+    case $arg in
+        --secondary=*)
+        secondary_count="${arg#*=}"
+        shift
+        ;;
+        *)
```
[the remark this answers is not in the document]
Comment (MykolaSolyanko) [reply scripts/create_aos_vm.sh]: Done

File: scripts/create_aos_vm.sh [hunk 7]
```diff
@@ -2,42 +2,90 @@
...
+
 if [ "$#" -lt 1 ]; then
-    echo "Expect 1 parameter. Usage example: ./${script} <image_path>"
+    print_usage
+    exit 1
+fi
+
+secondary_count=1
+
+for arg in "$@"; do
```
[the remark this answers is not in the document]
Comment (MykolaSolyanko) [reply scripts/create_aos_vm.sh]: Done

File: scripts/run_secondary_nodes.sh [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+    echo "Converting $node_image to vmdk..."
+    qemu-img convert -p -f raw -O vmdk "${node_image}" "$image_path/${vmdk_name}"
+    return 0
+}
+
+generate_mac() {
+    echo "52:54:00$(hexdump -n3 -e '":"1/1 "%02X"' /dev/urandom)"
+}
+
+start_node() {
```
[the remark this answers is not in the document]
Comment (MykolaSolyanko) [reply scripts/run_secondary_nodes.sh]: Done

File: README.md (hunk 1 above)
Comment (MykolaSolyanko) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (MykolaSolyanko) [reply README.md]: Done

File: README.md (hunk 3 above)
Comment (MykolaSolyanko) [reply README.md]: Done

**Review by al1img — APPROVED** [review]

**Pull request merged**